app/web/components/CustomColorSwitch.tsx

```diff
@@ -0,0 +1,84 @@
+import { Switch, withStyles } from "@material-ui/core";
+import makeStyles from "utils/makeStyles";
+
+import CircularProgress from "./CircularProgress";
+
+const useStyles = makeStyles(({ palette, shadows }) => ({
+  circle: {
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+    width: 20,
+    height: 20,
+    borderRadius: "50%",
+    backgroundColor: palette.grey[600],
+    boxShadow: shadows[1],
+  },
+  active: {
+    backgroundColor: palette.grey[600],
+  },
+}));
+
+export default function CustomColorSwitch({
+  checked,
+  color,
+  isLoading,
+  onClick,
+  status,
+}: {
+  checked: boolean;
+  color: string;
+  isLoading: boolean;
+  onClick: () => void;
+  status: string;
+}) {
+  const classes = useStyles();
+
+  const CustomColorSwitch = withStyles((theme) => ({
+    switchBase: {
+      color: theme.palette.grey[600],
+      "& + $track": {
+        backgroundColor: theme.palette.grey[200],
+      },
+      "&$checked": {
+        color,
+        "& + $track": {
+          backgroundColor: color,
+        },
+      },
+      "&$disabled": {
+        color: checked ? color : theme.palette.grey[600],
+        "& + $track": {
+          backgroundColor: checked ? color : theme.palette.grey[200],
+          opacity: 0.4,
+        },
+      },
+    },
+    checked: {},
+    disabled: {},
+    track: {},
+  }))(Switch);
```

Comment:
I am wondering about this: I've not seen it elsewhere in the codebase (using `withStyles`). I feel we have some other custom-styled Mui components as well. I'm not sure how they are done.

Another thing is that should we just apply this override globally in `theme.ts`? Is there a case where we don't want this custom styled switch?

Anyway, this is not so important, feel free to ignore :)

Reply:
I've changed it to using our `makeStyles` and switched to the custom-styled Mui components.

As for the global override, should I make a separate ticket for that? I can make sure all the existing switches use the loading state?

I originally did this as the existing switch was orange and it could only be made the primary, secondary or default colors, none of which we wanted based on the design.

I think better in a separate ticket then if I mess something up or miss one easier to notice and review.

Reply:
Sounds good!